test/src/Grant/GrantTestCase.php

```diff
@@ -50,36 +46,41 @@ protected function setUp()
      */
     public function testGetAccessToken($grant, array $params = [])
     {
-        // Mock
-        $stream = Phony::mock(StreamInterface::class);
-        $stream->__toString->returns(
-            '{"access_token": "mock_access_token", "expires": 3600, "refresh_token": "mock_refresh_token", "uid": 1}'
-        );
+        $provider = $this->getMockProvider();
 
-        $response = Phony::mock(ResponseInterface::class);
-        $response->getBody->returns($stream->get());
-        $response->getHeader->with('content-type')->returns('application/json');
+        /** @var StreamInterface & MockInterface $stream */
+        $stream = Mockery::spy(StreamInterface::class)->makePartial();
+        $stream
+            ->shouldReceive('__toString')
+            ->once()
+            ->andReturn('{"access_token": "mock_access_token", "expires": 3600, "refresh_token": "mock_refresh_token", "uid": 1}');
 
-        $client = Phony::mock(ClientInterface::class);
-        $client->send->returns($response->get());
+        /** @var ResponseInterface & MockInterface $response */
+        $response = Mockery::spy(ResponseInterface::class)->makePartial();
+        $response
+            ->shouldReceive('getBody')
+            ->once()
+            ->andReturn($stream);
+        $response
+            ->shouldReceive('getHeader')
+            ->once()
+            ->with('content-type')
+            ->andReturn('application/json');
 
-        // Execute
-        $this->provider->setHttpClient($client->get());
-        $token = $this->provider->getAccessToken($grant, $params);
+        /** @var ClientInterface & MockInterface $client */
+        $client = Mockery::spy(ClientInterface::class)->makePartial();
+        $client
+            ->shouldReceive('send')
+            ->once()
+            ->withArgs(function ($request) {
+                parse_str((string) $request->getBody(), $body);
+                return call_user_func($this->getParamExpectation(), $body);
+            })
+            ->andReturn($response);
```

Comment:
Slightly disappointed at the loss of Phony, I find it far more elegant.

Reply:
Sorry. It appears that it generates code and `eval()`s it to create mocks, and the code it generated kept putting a backslash in front of `?string` type hints, causing syntax errors (i.e. `\?string`). I spent a fair amount of time trying to juggle dependencies to see if I could find the right combination that worked across all PHP versions.

In the end, I decided to change oauth2-client to get it passing on all versions of PHP, so that we could announce PHP 8 support as soon as PHP 8 is out. When Phony supports newer versions of PHP, we can switch back. Switching everything to Mockery didn't take more than an hour, so switching back shouldn't be a big deal.